src/SonarScanner.MSBuild.PreProcessor/TargetsInstaller.cs

```diff
@@ -96,26 +96,36 @@ private void CopyIfDifferent(string sourcePath, IEnumerable<string> destinationD
             {
                 var destinationPath = Path.Combine(destinationDir, fileName);
 
-                if (!this.fileWrapper.Exists(destinationPath))
+                try
                 {
-                    this.directoryWrapper.CreateDirectory(destinationDir); // creates all the directories in the path if needed
-                    this.fileWrapper.Copy(sourcePath, destinationPath, overwrite: false);
-                    this.logger.LogDebug(Resources.MSG_InstallTargets_Copy, fileName, destinationDir);
-                }
-                else
-                {
-                    var destinationContent = this.fileWrapper.ReadAllText(destinationPath);
-
-                    if (!string.Equals(sourceContent, destinationContent, StringComparison.Ordinal))
+                    if (!this.fileWrapper.Exists(destinationPath))
                     {
+                        this.directoryWrapper.CreateDirectory(destinationDir); // creates all the directories in the path if needed
+
+                        // always overwrite to avoid intermittent exceptions: https://github.com/SonarSource/sonar-scanner-msbuild/issues/647
                         this.fileWrapper.Copy(sourcePath, destinationPath, overwrite: true);
-                        this.logger.LogDebug(Resources.MSG_InstallTargets_Overwrite, fileName, destinationDir);
+                        this.logger.LogDebug(Resources.MSG_InstallTargets_Copy, fileName, destinationDir);
                     }
                     else
                     {
-                        this.logger.LogDebug(Resources.MSG_InstallTargets_UpToDate, fileName, destinationDir);
+                        var destinationContent = this.fileWrapper.ReadAllText(destinationPath);
+
+                        if (!string.Equals(sourceContent, destinationContent, StringComparison.Ordinal))
+                        {
+                            this.fileWrapper.Copy(sourcePath, destinationPath, overwrite: true);
+                            this.logger.LogDebug(Resources.MSG_InstallTargets_Overwrite, fileName, destinationDir);
+                        }
+                        else
+                        {
+                            this.logger.LogDebug(Resources.MSG_InstallTargets_UpToDate, fileName, destinationDir);
+                        }
                     }
                 }
+                catch (Exception e)
+                {
+                    this.logger.LogWarning(Resources.MSG_InstallTargets_Error, destinationPath, e.Message);
+                    this.logger.LogDebug(e.StackTrace);
```

Comment:
Adding logging in case an exception is thrown - for example if a file is locked or we don't have access to read/write.